### DISCUSSION TITLE
Does NGitLab currently support the Generic Packages Repository

### DISCUSSION THREAD

**Discussion by gep13** [ubisoft/NGitLab#511] [Q&A]
What I am referring to is this:

https://docs.gitlab.com/ee/user/packages/generic_packages/index.html#publish-a-package-file

I don't see an obvious client that could be used to upload a file into this repository, but I was hoping that I am missing something.

Thanks

**Comment by meziantou** [comment]
NGitLab does not support package endpoints. Feel You can submit a PR if you want.

**Reply by gep13** [reply]
Happy to try to submit a PR for this, but as yet, I haven't looked at the source code for NGitLab to figure out how it works under the hood. Would you be able to provide any guidance/directions on where this new functionality would go? Thanks!

**Reply by meziantou** [reply]
You should:
- Add a new interface `IPackageClient` to expose the new methods
- Add a new property or method in `IGitLabClient` to get an instance of `IPackageClient`

Depending on what should be in the request body, you may also need to update https://github.com/ubisoft/NGitLab/blob/fde8c62576f82c4df09a6584edcc48fb59d3c3f0/NGitLab/Impl/HttpRequestor.GitLabRequest.cs#L155-L183.

You don't need to implement the interface in the mock project. You can leave the `NotImplementedException`.

To run the tests locally, you can check the readme https://github.com/ubisoft/NGitLab?tab=readme-ov-file#running-unit-tests-locally
